modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java

```diff
@@ -531,33 +531,37 @@ public void run() {
         StatusThreadPoolExecutor.getInstance().execute(r);
     }
 
-    private void executeRunnableStatusGoMethod(Supplier<String> method, Callback callback) throws JSONException {
+    private void executeRunnableStatusGoMethod(Supplier<String> method, Callback successCallback,Callback errorCallback) throws JSONException {
         if (!checkAvailability()) {
-            callback.invoke(false);
+            successCallback.invoke(false);
             return;
         }
 
         Runnable runnableTask = () -> {
             String res = method.get();
-            callback.invoke(res);
+             if (res.contains("error")) {
```

Comment:
This is very dangerous, what if you receive a message with `error` in the content? we need to parse it as JSON (could be optimized by scanning the string, but don't think it's worth the risk)

Reply:
hmmm, every method in status-go uses `makeJSONResponse ` method 
which will add the error key to the response string :  reference https://github.com/status-im/status-go/blob/ec7c0e9c7d34f73d89d8fb4be0e116aa29c3a29d/mobile/status.go#L625

Reply:
> This is very dangerous, what if you receive a message with `error` in the content? we need to parse it as JSON (could be optimized by scanning the string, but don't think it's worth the risk)

Hmmm I see your point

Reply:
I've done this on Android native module, I will work on the IOS side as well! Thanks for the suggestion @cammellos

Reply:
> This is very dangerous, what if you receive a message with `error` in the content? we need to parse it as JSON (could be optimized by scanning the string, but don't think it's worth the risk)

For ios side I converted the response to a dictionary and used a similar approach here, this should ensure we look for the error key and we do not trigger false positives by "error" in message